conans/model/graph_lock.py

```diff
@@ -146,6 +149,9 @@ def merge(self, other):
         self._requires.merge(other._requires)
         self._build_requires.merge(other._build_requires)
         self._python_requires.merge(other._python_requires)
+        # TODO: Check if this holds for merges
+        self._alias.update(other._alias)
```

Comment:
I think this is a bugfix for `lock merge` too no? Maybe we should add a changelog line for it too if so. (After looking at the code, I haven't much `merge` tests which would have caught this so I'm now not sure)

Reply:
Agree, doing a separate PR for it in https://github.com/conan-io/conan/pull/13763, with its own test